packages/analytics/analytics-geo-map/src/components/AnalyticsGeoMap.vue

```diff
@@ -0,0 +1,244 @@
+    ref="mapContainer"
+    class="kong-ui-public-analytics-geo-map"
+  />
+</template>
+
+<script setup lang="ts">
+import { ref, onMounted, watch, computed } from 'vue'
+import { Map, Popup } from 'maplibre-gl'
+import type { PropType } from 'vue'
+import type { ColorSpecification, DataDrivenPropertyValueSpecification, ExpressionSpecification, LngLatLike, MapOptions } from 'maplibre-gl'
+import type { LongLat, MapFeatureCollection } from '../types'
+import type { FeatureCollection, Feature, MultiPolygon } from 'geojson'
+import 'maplibre-gl/dist/maplibre-gl.css'
+
+const props = defineProps({
+  countryMetrics: {
+    type: Object as PropType<Record<string, number>>,
+    required: true,
+  },
+  geoJsonData: {
+    type: Object as PropType<MapFeatureCollection>,
+    required: true,
+  },
+  center: {
+    type: Object as PropType<LongLat>,
+    required: false,
+    default: null,
+  },
+  fitToCountry: {
+    type: String,
+    required: false,
+    default: '',
+  },
+  initialZoom: {
+    type: Number,
+    required: false,
+    default: null,
+    validator: (value: number) => value >= 0 && value <= 24,
+  },
+  metricUnit: {
+    type: String,
+    required: false,
+    default: '',
+  },
+})
+
+const mapContainer = ref<HTMLDivElement | null>(null)
+const map = ref<Map>()
+
+const layerPaint = computed(() => ({
+  'fill-color': [
+    'match',
+    ['get', 'ISO_A2'] as ExpressionSpecification,
+    ...(Object.keys(props.countryMetrics).length
+      ? Object.entries(props.countryMetrics).flatMap(([code, metric]) => [
+        code,
+        getColor(metric),
+      ])
+      : ['#ccc', '#ccc']),
+    '#ccc', // default color if no match
+  ] as DataDrivenPropertyValueSpecification<ColorSpecification>,
+  'fill-opacity': 0.7,
+}))
+
+const getColor = (metric: number) => {
+  if (metric > 80) return '#296378'
+  if (metric > 60) return '#0D8093'
+  if (metric > 40) return '#009FA9'
+  if (metric > 20) return '#00BDB7'
+  return '#0CDCBD'
+}
+
+// Simplified coords may be 2 or 3 layers
+const flattenPositions = (position: any): number[][] => {
+  const flat: (number | number[])[] = position.flat(1)
+
+  if (Array.isArray(flat[0]) && Array.isArray(flat[0][0])) {
+    return flattenPositions(flat)
+  }
+
+  return flat as number[][]
+}
+
+const goToCountry = (countryCode: string) => {
+
+  // Overrides for large spanning countries
+  if (countryCode === 'RU') {
+    map.value?.fitBounds([
+      [20, 40],
+      [180, 80],
+    ])
+    return
+  } else if (countryCode === 'US') {
+    map.value?.fitBounds([
+      [-130, 20],
+      [-60, 50],
+    ])
+    return
+  } else if (countryCode === 'FR') {
+    map.value?.fitBounds([
+      [-5, 42],
+      [9, 51],
+    ])
+    return
+  } else if (countryCode === 'NO') {
+    map.value?.fitBounds([
+      [4, 57],
+      [32, 71],
+    ])
+    return
+  }
+
+  const found = (props.geoJsonData as FeatureCollection).features.find((f: Feature) => f.properties?.ISO_A2 === countryCode)
```

Comment:
Couple things:

1. Can you type the prop instead of force casting it?
2. Can you type the `found` variable? I think it’s a boolean
3. Should it be computed? If the prop changes after this is initialized it will be static

Reply:
done